### PR TITLE
Add support for hls streams in bigo

### PR DESCRIPTION
**Pull request by unnutricious**
This is a workaround for issue #1031 since FLV streaming is broken

### CODE REVIEW

File: src/streamlink/plugins/bigo.py [hunk 1]
```diff
@@ -67,5 +70,15 @@ def _get_streams(self):
         sid, uid, ip, port = flashvars.groups()
         yield "live", BigoStream(self.session, sid, uid, ip, port)
 
+        page = http.get(self.url,
+                        allow_redirects=True,
+                        headers={"User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 8_0_2 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Version/8.0 Mobile/12A366 Safari/600.1.4"})
```
Comment (beardypig) [inline src/streamlink/plugins/bigo.py]: There is an iPhone user agent const that you can use `streamlink.plugin.api.useragents`.

File: src/streamlink/plugins/bigo.py [hunk 2]
```diff
@@ -67,5 +70,15 @@ def _get_streams(self):
...
+        videomatch = self._video_re.search(page.text)
+        if not videomatch:
+            return
+
+        videourl = videomatch.group(1)
+        yield "hls", HLSStream(self.session, videourl)
```
Comment (beardypig) [inline src/streamlink/plugins/bigo.py]: If you name the stream `live` won't it be `best` by default?

File: src/streamlink/plugins/bigo.py (hunk 1 above)
Comment (beardypig) [inline src/streamlink/plugins/bigo.py]: You could update the other http call too as it was written before that module was added :-)

**Comment by unnutricious** [comment]
Addressed comments and also improved the url regex

File: src/streamlink/plugins/bigo.py [hunk 3]
```diff
@@ -59,13 +62,23 @@ def can_handle_url(cls, url):
...
                         allow_redirects=True,
-                        headers={"User-Agent": "Mozilla/5.0 (MSIE 10.0; Windows NT 6.1; Trident/5.0)"})
+                        headers={"User-Agent": useragents.CHROME})
         flashvars = self._flashvars_re.search(page.text)
         if not flashvars:
             return
 
         sid, uid, ip, port = flashvars.groups()
-        yield "live", BigoStream(self.session, sid, uid, ip, port)
+        yield "flv", BigoStream(self.session, sid, uid, ip, port)
```
Comment (beardypig) [inline src/streamlink/plugins/bigo.py]: Doesn't it work if both streams are called `live`? HLS shout have a higher priority.

File: src/streamlink/plugins/bigo.py [hunk 4]
```diff
@@ -47,10 +47,13 @@ def close(self):
 
 
 class Bigo(Plugin):
-    _url_re = re.compile(r"https?://(?:www\.)?(bigo\.tv/\d+|bigoweb\.co/show/\d+)")
+    _url_re = re.compile(r"http://(www.bigo.tv/[\w\d]+|bigoweb.co/show/\d+)")
```
Comment (beardypig) [inline src/streamlink/plugins/bigo.py]: This won't match www.bigweb.co or https anymore, did you intend to do that?

**Comment by bigotvuser** [comment]
Although fully agree that improving the URL regex is being done at an opportune moment, could just the HLS changes be released for now so that it can be tested. It appears the change is regex would apply to only a minority of bigo tv URLs.

**Comment by beardypig** [comment]
Currently the tests are not passing, so this cannot be merged in its current state.

**Comment by bastimeyer** [comment]
Since @unnutricious seems to be unavailable right now, someone could pull their changes, squash all the commits, add the needed fix and open a new PR with these two commits. This way, everyone will be credited for their work and the plugin gets fixed.

**Comment by unnutricious** [comment]
Sorry, the pr should be in better shape now. I removed flv support because it probably won't work again.

**Comment by beardypig** [comment]
@unnutricious, nice looks good to me :)

**Comment by gravyboat** [comment]
Thanks @unnutricious! Thanks for the review @beardypig!

**Pull request merged**